### PR TITLE
MWPW-142699 allowing empty defined value for PH

### PR DESCRIPTION
**Pull request by npeltier**
problem and solution discussed in:
https://github.com/orgs/adobecom/discussions/1900

Resolves: [MWPW-142699](https://jira.corp.adobe.com/browse/MWPW-142699)

**Test URLs:**
- Before: https://main--milo--adobecom.hlx.page/drafts/npeltier/placeholder-tests?martech=off
- After: https://mwpw-142699--milo--npeltier.hlx.page/drafts/npeltier/placeholder-tests?martech=off

### CODE REVIEW

**Review by mokimo — APPROVED** [review]

File: test/features/placeholders/placeholders.test.js [hunk 1]
```diff
@@ -35,4 +35,9 @@ describe('Placeholders', () => {
     const text = await replaceKey('this-wont-work', config);
     expect(text).to.equal('this wont work');
   });
+
+  it('Do show empty value', async () => {
```
Comment (mokimo) [inline test/features/placeholders/placeholders.test.js:39]: Wording nit: (it) `Does show an empty value`

File: libs/features/placeholders.js [hunk 2]
```diff
@@ -69,7 +69,7 @@ async function getPlaceholder(key, config, sheet) {
     return defaultPlaceholders;
   });
 
-  if (placeholders?.[key]) return placeholders[key];
+  if (typeof placeholders?.[key] !== 'undefined') return placeholders[key];
```
Comment (overmyheadandbody) [inline libs/features/placeholders.js:72]: Placeholder values should be strings, regardless of their length, so would it make sense to instead check for `typeof placeholders?.[key] === 'string'`?

Comment (mokimo) [reply libs/features/placeholders.js:72]: I was wondering the same, I'm not 100% certain on how EDS handles this and I _think_ values might be `null` which would still mean the key exists but the value is omitted

**Review by 3ch023 — APPROVED** [review]

**Review by robert-bogos — APPROVED** [review]

**Review by sukamat — APPROVED** [review]

**Review by honstar — APPROVED** [review]

**Comment by afmicka** [comment]
@npeltier This branch seems to be braking all the pages with blocks. Could you please check on that?
CC: https://main--cc--adobecom.hlx.live/creativecloud/file-types/image/comparison?milolibs=mwpw-142699--milo--npeltier&martech=off
DC: https://main--dc--adobecom.hlx.live/acrobat/resources/how-to-write-an-ebook?milolibs=mwpw-142699--milo--npeltier&martech=off
bacom: https://main--bacom--adobecom.hlx.page/products/adobe-experience-cloud-products?milolibs=mwpw-142699--milo--npeltier&martech=off
kitchen-sink: https://mwpw-142699--milo--npeltier.hlx.page/docs/library/kitchen-sink/merch-card
Screenshot 2024-02-22 at 10 59 45

**Comment by npeltier** [comment]
@afmicka i fixed it i think (was codebus cache issue)

**Comment by afmicka** [comment]
@npeltier Yes, it is fixed now. Will verify the PR.

**Pull request merged**